apicast/src/proxy.lua

```diff
@@ -213,8 +222,13 @@ function _M:authorize(service, usage, credentials, ttl)
 
     local res = http.get(internal_location)
 
-    if not self:handle_backend_response(cached_key, res, ttl) then
-      error_authorization_failed(service)
+    local authorized, rejection_reason = self:handle_backend_response(cached_key, res, ttl)
+    if not authorized then
+      if rejection_reason == 'limits_exceeded' then
+        error_limits_exceeded(service)
```

Comment:
Can you add `return` to both branches, so it is obvious that nothing is executed after (because it calls `ngx.exit`)?

Reply:
Good idea to make it more explicit. Changed :+1: